api/placer.py

```diff
@@ -0,0 +1,278 @@
+        minimum = datetime.datetime(1980, 1, 1).isoformat()
+        stamp   = self.metadata['acquisition'].get('timestamp', minimum)
+        self.ziptime = int(dateutil.parser.parse(stamp).strftime('%s'))
+
+        # The zipfile is a santizied acquisition label
+        self.dir = util.sanitize_string_to_filename(self.a_label)
+        self.name = self.dir + '.zip'
+
+        # Make a tempdir to store zip until moved
+        # OPPORTUNITY: this is also called in files.py. Could be a util func.
+        self.tempdir = tempfile.TemporaryDirectory(prefix='.tmp', dir=config.get_item('persistent', 'data_path'))
+
+        # Create a zip in the tempdir that later gets moved into the CAS.
+        self.path = os.path.join(self.tempdir.name, 'temp.zip')
+        self.zip  = zipfile.ZipFile(self.path, 'w', zipfile.ZIP_DEFLATED, allowZip64=True)
+
+        # OPPORTUNITY: add zip comment
+        # self.zip.comment = json.dumps(metadata, default=metadata_encoder)
+
+        # Bit of a silly hack: write our tempdir directory into the zip (not including its contents).
+        # Creates an empty directory entry in the zip which will hold all the files inside.
+        # This way, when you expand a zip, you'll get folder/things instead of a thousand dicoms splattered everywhere.
+        self.zip.write(self.tempdir.name, self.dir)
+
+    def process_file_field(self, field, info):
+        # Set the file's mtime & atime.
+        os.utime(field.path, (self.ziptime, self.ziptime))
+
+        # Place file into the zip folder we created before
+        self.zip.write(field.path, os.path.join(self.dir, field.filename))
+
+    def finalize(self):
+        self.zip.close()
+
+        # Create an anyonmous object in the style of our augmented file fields.
+        # Not a great practice. See process_upload() for details.
+        cgi_field = util.obj_from_map({
+            'filename': self.name,
+            'path':	 self.path,
+            'size':	 os.path.getsize(self.path),
+            'hash':	 files.hash_file_formatted(self.path),
+            'mimetype': util.guess_mimetype('lol.zip'),
+            'modified': self.timestamp
+        })
+
+        # Similarly, create the info map that is consumed by helper funcs. Clear duplication :(
+        # This could be coalesced into a single map thrown on file fields, for example.
+        # Used in the API return.
+        cgi_info = {
+            'name':	 cgi_field.filename,
+            'modified': cgi_field.modified,
+            'size':	 cgi_field.size,
+            'hash':	 cgi_field.hash,
+
+            'type': self.metadata['packfile']['type'],
+
+            # OPPORTUNITY: packfile endpoint could be extended someday to take additional metadata.
+            'instrument': None,
+            'measurements': [],
+            'tags': [],
+            'metadata': {}
+        }
+
+        # Get or create a session based on the hierarchy and provided labels.
+        s = {
```

Comment:
yeah you need to get the group id from the project object. The input session schema doesn't have the group so that you just need the parent project to create a session inside it. I don't remember the reason why we need to store the group id in the session (Cc @gsfr) but it is a required field.

The acquisition doesn't need any other key.